### PR TITLE
Add BedJet Fan child component

### PR DESCRIPTION
**Pull request by jhansche**
# What does this implement/fix?

This adds `fan` domain child component to the `bedjet` platform/hub.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#2239

## Test Environment

- [X] ESP32
- [X] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:

```diff
diff --git config.yaml config.yaml
@@ -1,12 +1,18 @@
 # Example config.yaml
 
 esp32_ble_tracker:
 
 ble_client:
   - mac_address: C4:4F:33:00:00:01
     id: bedjet_ble_id1

 bedjet:
   - id: bedjet_1
     ble_client_id: bedjet_ble_id1

+fan:
+  - platform: bedjet
+    id: my_bedjet_fan_entity
+    name: "My BedJet fan"
+    bedjet_id: bedjet_1

```

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/bedjet/climate/__init__.py [hunk 1]
```diff
@@ -9,19 +9,17 @@
...
     BEDJET_CLIENT_SCHEMA,
     register_bedjet_child,
 )
 
 _LOGGER = logging.getLogger(__name__)
 CODEOWNERS = ["@jhansche"]
-DEPENDENCIES = ["ble_client"]
+DEPENDENCIES = ["bedjet"]
 
 bedjet_ns = cg.esphome_ns.namespace("bedjet")
```
Comment (jesserockz) [inline esphome/components/bedjet/climate/__init__.py]: This should be imported from the parent file too like `BEDJET_CLIENT_SCHEMA`

File: esphome/components/bedjet/fan/__init__.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+from .. import (
+    BEDJET_CLIENT_SCHEMA,
+    register_bedjet_child,
+)
+
+_LOGGER = logging.getLogger(__name__)
+CODEOWNERS = ["@jhansche"]
+DEPENDENCIES = ["bedjet"]
+
+bedjet_ns = cg.esphome_ns.namespace("bedjet")
```
Comment (jesserockz) [inline esphome/components/bedjet/fan/__init__.py]: As above, import from parent.

**Review by jesserockz — APPROVED** [review]
Looks good, just two non-critical changes please

File: esphome/components/bedjet/climate/__init__.py (hunk 1 above)
Comment (jhansche) [reply esphome/components/bedjet/climate/__init__.py]: Done

**Review by jesserockz — APPROVED** [review]

**Pull request merged**